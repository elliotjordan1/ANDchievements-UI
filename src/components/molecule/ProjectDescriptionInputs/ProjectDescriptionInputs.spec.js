import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Component from '.';

describe('ProjectDescriptionInputs', () => {
  it('renders successfully with base component', () => {
    const handleChange = jest.fn();
    const handleBlur = jest.fn();

    const values = {
      projectDescription: '',
      projetOutcomes: '',
      clientDescription: ''
    }

    const errors = {};

    const touched = {};

    const component = render(<Component
      handleChange={handleChange}
      handleBlur={handleBlur}
      values={values}
      errors={errors}
      touched={touched} />);

    expect(component).toMatchSnapshot();
  });
  it('renders successfully with errored FormLabels', () => {
    const handleChange = jest.fn();
    const handleBlur = jest.fn();

    const values = {
      projectDescription: '',
      projetOutcomes: '',
      clientDescription: ''
    }

    const errors = {
      projectDescription: true
    };

    const touched = {
      projectDescription: true,
      projectOutcomes: true,
      clientDescription: true
    };

    const component = render(<Component
      handleChange={handleChange}
      handleBlur={handleBlur}
      values={values}
      errors={errors}
      touched={touched} />);

    expect(component).toMatchSnapshot();
  });
  it('successfully fires onChange handler when text is changed', () => {
    const handleChange = jest.fn();
    const handleBlur = jest.fn();

    const values = {
      projectDescription: '',
      projetOutcomes: '',
      clientDescription: ''
    }

    const errors = {};

    const touched = {};

    const { getByPlaceholderText } = render(<Component
      handleChange={handleChange}
      handleBlur={handleBlur}
      values={values}
      errors={errors}
      touched={touched} />);

    const clientDescription = getByPlaceholderText('Client Description');
    const projectDescription = getByPlaceholderText('Project Description');
    const projectOutcomes = getByPlaceholderText('Project Outcomes');

    expect(handleChange).toHaveBeenCalledTimes(0);

    fireEvent.change(clientDescription, { target: { value: 'Test Text' }});
    fireEvent.change(projectDescription, { target: { value: 'Project Outcome' }});
    fireEvent.change(projectOutcomes, { target: { value: 'Outcome' }});

    expect(handleChange).toHaveBeenCalledTimes(3);
  })
})