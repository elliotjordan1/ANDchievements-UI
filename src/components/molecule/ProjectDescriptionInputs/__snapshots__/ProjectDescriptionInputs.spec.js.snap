// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDescriptionInputs renders successfully with base component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="FormLabel-wqrwj6-0 JaPpn"
      >
        Project Description
      </h2>
      <div
        class="InputContainer-sc-1hosyg-0 RYaOb"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="clientDescription"
          placeholder="Client Description"
          type="text"
          value=""
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="projectDescription"
          placeholder="Project Description"
          type="text"
          value=""
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="projectOutcomes"
          placeholder="Project Outcomes"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="FormLabel-wqrwj6-0 JaPpn"
    >
      Project Description
    </h2>
    <div
      class="InputContainer-sc-1hosyg-0 RYaOb"
    >
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="clientDescription"
        placeholder="Client Description"
        type="text"
        value=""
      />
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="projectDescription"
        placeholder="Project Description"
        type="text"
        value=""
      />
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="projectOutcomes"
        placeholder="Project Outcomes"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProjectDescriptionInputs renders successfully with errored FormLabels 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="FormLabel-wqrwj6-0 dPjrlO"
      >
        Project Description
      </h2>
      <div
        class="InputContainer-sc-1hosyg-0 RYaOb"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="clientDescription"
          placeholder="Client Description"
          type="text"
          value=""
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="projectDescription"
          placeholder="Project Description"
          type="text"
          value=""
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
          maxlength="60"
          name="projectOutcomes"
          placeholder="Project Outcomes"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="FormLabel-wqrwj6-0 dPjrlO"
    >
      Project Description
    </h2>
    <div
      class="InputContainer-sc-1hosyg-0 RYaOb"
    >
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="clientDescription"
        placeholder="Client Description"
        type="text"
        value=""
      />
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="projectDescription"
        placeholder="Project Description"
        type="text"
        value=""
      />
      <input
        class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
        maxlength="60"
        name="projectOutcomes"
        placeholder="Project Outcomes"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
